.bumpversion.toml

```diff
@@ -24,34 +24,48 @@ commit = true
 message = "Bump version: {current_version} → {new_version}"
 commit_args = ""
 
+# Java maven files
+pre_commit_hooks = [
+  """
+    NEW_VERSION="${BVHOOK_NEW_MAJOR}.${BVHOOK_NEW_MINOR}.${BVHOOK_NEW_PATCH}"
+    if [ ! -z "$BVHOOK_NEW_PRE_L" ] && [ ! -z "$BVHOOK_NEW_PRE_N" ]; then
+        NEW_VERSION="${NEW_VERSION}-${BVHOOK_NEW_PRE_L}.${BVHOOK_NEW_PRE_N}"
+    fi
+    echo "Constructed new version: $NEW_VERSION"
+    cd java && mvn versions:set -DnewVersion=$NEW_VERSION && mvn versions:commit
+    cd ..
+    git add java/
```

Comment:
Hmm that seems like it could be a bit dangerous. Someone might have intentionally unstaged changes in that directory. I feel like most commit hooks would just detect if anything changed and then error if it did, right? That would give the developer the chance to review and add the changes themself.

Reply:
Good catch! Agree that it could be error-prone. modified the script to detect changes to pom.xml files and add those files. And ensure we can get information of what happened